aws/resource_aws_instance.go

```diff
@@ -1835,22 +1856,22 @@ func readVolumeTags(conn *ec2.EC2, instanceId string) ([]*ec2.Tag, error) {
 // config determine which one to use in Plan and Apply.
 func readSecurityGroups(d *schema.ResourceData, instance *ec2.Instance, conn *ec2.EC2) error {
 	// An instance with a subnet is in a VPC; an instance without a subnet is in EC2-Classic.
-	hasSubnet := instance.SubnetId != nil && *instance.SubnetId != ""
+	hasSubnet := instance.SubnetId != nil && aws.StringValue(instance.SubnetId) != ""
```

Comment:
`aws.StringValue(instance.SubnetId) != ""` will be sufficient as `aws.StringValue` converts `nil` to `""`.